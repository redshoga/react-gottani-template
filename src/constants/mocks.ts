import { Post } from "../models/Post";

const mockPosts: Post[] = [
  {
    userId: 1,
    id: 1,
    title: "hello post1",
    body: "body test1"
  },
  {
    userId: 2,
    id: 2,
    title: "hello post2",
    body: "body test2"
  }
];

export { mockPosts };
