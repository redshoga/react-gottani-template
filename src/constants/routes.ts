export const ROUTES = {
  ROOT: "/",
  COUNTER: "/counter",
  LOGO: "/logo",
  POSTS: "/posts"
};
