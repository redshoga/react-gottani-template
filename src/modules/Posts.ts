import { Action } from "redux";

// action types
export const FETCH_POSTS = "FETCH_POSTS";
export const SET_POSTS = "SET_POSTS";
export const POSTS_FETCH_FAILED = "POSTS_FETCH_FAILED";

interface FetchPostsAction extends Action {
  type: typeof FETCH_POSTS;
}

interface SetPostsAction extends Action {
  type: typeof SET_POSTS;
  payload: Post[];
}

interface PostsFetchFailedAction extends Action {
  type: typeof POSTS_FETCH_FAILED;
}

export type ActionTypes =
  | FetchPostsAction
  | SetPostsAction
  | PostsFetchFailedAction;

// action creator
const fetchPosts: () => ActionTypes = () => {
  return { type: FETCH_POSTS };
};
const setPosts: (posts: Post[]) => ActionTypes = posts => {
  return {
    type: SET_POSTS,
    payload: posts
  };
};
const postsFetchFailed: () => ActionTypes = () => {
  return { type: POSTS_FETCH_FAILED };
};

export const actionCreator = {
  fetchPosts,
  setPosts,
  postsFetchFailed
};

// state
export interface Post {
  userId: number;
  id: number;
  title: string;
  body: string;
}
export interface State {
  posts: Post[];
  loadingPosts: boolean;
  fetchFailed: boolean;
}
export const initialState: State = {
  posts: [],
  loadingPosts: false,
  fetchFailed: false
};

// reducer
export function reducer(state = initialState, action: ActionTypes): State {
  switch (action.type) {
    case FETCH_POSTS:
      return {
        ...state,
        loadingPosts: true
      };
    case SET_POSTS:
      return {
        ...state,
        posts: action.payload,
        fetchFailed: false,
        loadingPosts: false
      };
    case POSTS_FETCH_FAILED:
      return {
        ...state,
        fetchFailed: true
      };
    default:
      return state;
  }
}
