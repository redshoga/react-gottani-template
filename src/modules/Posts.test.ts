import reducerTester from "reducer-tester";
import { reducer, initialState, actionCreator } from "./Posts";
import { mockPosts } from "../constants/mocks";

describe("Counter", () => {
  reducerTester({
    reducer,
    state: initialState,
    tests: [
      actionCreator.fetchPosts(),
      actionCreator.postsFetchFailed(),
      actionCreator.setPosts(mockPosts)
    ]
  });
});
