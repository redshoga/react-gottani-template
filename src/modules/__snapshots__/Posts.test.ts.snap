// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Posts/FETCH_POSTS 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchFailed": false,
-   "loadingPosts": false,
+   "loadingPosts": true,
    "posts": Array [],
  }
`;

exports[`Counter Posts/POSTS_FETCH_FAILED 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
-   "fetchFailed": false,
+   "fetchFailed": true,
    "loadingPosts": false,
    "posts": Array [],
  }
`;

exports[`Counter Posts/SET_POSTS 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
    "fetchFailed": false,
    "loadingPosts": false,
-   "posts": Array [],
+   "posts": Array [
+     Object {
+       "body": "body test1",
+       "id": 1,
+       "title": "hello post1",
+       "userId": 1,
+     },
+     Object {
+       "body": "body test2",
+       "id": 2,
+       "title": "hello post2",
+       "userId": 2,
+     },
+   ],
  }
`;

exports[`Counter handle initial state 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
